### PR TITLE
Remove unnecessary redeclarations of LINE_JOINER

### PR DESCRIPTION
**Pull request by Dominator008**
[Review on Reviewable](https://reviewable.io/reviews/google/closure-compiler/1588)

### CODE REVIEW

File: test/com/google/javascript/jscomp/TypeCheckTest.java [hunk 1]
```diff
@@ -43,7 +43,6 @@
 
 public final class TypeCheckTest extends CompilerTypeTestCase {
 
-  private static final Joiner LINE_JOINER = Joiner.on("\n");
   private static final String SUGGESTION_CLASS =
```
Comment (Dominator008) [inline test/com/google/javascript/jscomp/TypeCheckTest.java]: Is this `SUGGESTION_CLASS` used internally? It doesn't seem to be used externally so I can remove it as well.

**Comment by MatrixFrog** [comment]
It looks like SUGGESTION_CLASS is used internally, yes. As is the LINE_JOINER in TypeCheckTest. Not sure about the other ones...

File: test/com/google/javascript/jscomp/TypeCheckTest.java [hunk 2]
```diff
@@ -4043,7 +4042,7 @@ public void testInterfaceExtends() {
   }
 
   public void testDontCrashOnDupPropDefinition() {
-    testTypes(Joiner.on('\n').join(
+    testTypes(LINE_JOINER.join(
```
Comment (MatrixFrog) [inline test/com/google/javascript/jscomp/TypeCheckTest.java]: +1 to this. But then of course you'd need to undo the change on line 46.

Comment (Dominator008) [inline test/com/google/javascript/jscomp/TypeCheckTest.java]: https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/rhino/testing/BaseJSTypeTestCase.java#L58
already has a LINE_JOINER defined, so the one in `TypeCheckTest` was shadowing that and seems unnecessary.

File: test/com/google/javascript/jscomp/PolymerBehaviorExtractorTest.java [hunk 3]
```diff
@@ -28,8 +27,6 @@
  */
 public class PolymerBehaviorExtractorTest extends CompilerTypeTestCase {
 
-  private static final Joiner LINE_JOINER = Joiner.on('\n');
```
Comment (MatrixFrog) [inline test/com/google/javascript/jscomp/PolymerBehaviorExtractorTest.java]: This is used, but we could rename the one in CompilerTypeTestCase and make it protected.

Comment (Dominator008) [inline test/com/google/javascript/jscomp/PolymerBehaviorExtractorTest.java]: You mean the one in `BaseJSTypeTestCase`? It's already named `LINE_JOINER` and is protected: https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/rhino/testing/BaseJSTypeTestCase.java#L58
I thought Polymer*Test should just use that one instead?

Comment (MatrixFrog) [inline test/com/google/javascript/jscomp/PolymerBehaviorExtractorTest.java]: Ah, I missed that one. They should yes. But then let's get rid of this one because it's confusing to have both: https://github.com/google/closure-compiler/blob/b0b676c18657f6d8590c79de47c625e170ebae3d/test/com/google/javascript/jscomp/CompilerTypeTestCase.java#L69

Comment (Dominator008) [inline test/com/google/javascript/jscomp/PolymerBehaviorExtractorTest.java]: Done.

**Comment by Dominator008** [comment]
Closed by https://github.com/google/closure-compiler/commit/819c6099c1dc27c0a8612695f372c0abc4fa82a6

**Comment by Dominator008** [comment]
Closed by https://github.com/google/closure-compiler/commit/4397969a73b5fac74de6bd0501b4bbb44641c5df
